src/ovro_lwa_portal/io.py

```
@@ -84,61 +81,40 @@ def _resolve_doi(doi: str) -> str:
+            if url:
+                logger.info(f"Resolved DOI to zarr URL: {url}")
+                return url
+
+        # Fallback to first available URL if no zarr media type
+        url = download_url(doi)
+        if url:
+            logger.warning(f"No zarr media type found, using first URL: {url}")
+            return url
+
+        msg = f"No download URL found for DOI: {doi}"
+        raise DataSourceError(msg)
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)